### DISCUSSION TITLE
After modifying the original constant to a variable, the optimization solution can not get the result.

### DISCUSSION THREAD

**Discussion by LiuJincsu** [yalmip/YALMIP#1294] [Optimization modelling]
I'm working on an integrated energy system optimization problem. This problem involves the optimization of the capacity of multiple devices, including the capacity optimization of energy storage devices. When I set the energy storage capacity to constant, it can be solved quickly and successfully. But when I set storage capacity as a variable, since capacity is a time-independent variable, I set TEScap=sdpvar(1,1'full'). Moreover, TEScap is constrained by the upper and lower limits of capacity. But after this change can not solve the result, the page is as follows:
![image](https://github.com/yalmip/YALMIP/assets/100891235/83ca0093-ef4e-4939-8178-0ca31ab2dcf5)

**Comment by johanlofberg** [comment]
The display tells you that you have started the solution of a mixed-integer nonlinear program since the built-in solver for that has been started. If this is not 100% intentional and what you expect, you have problems. The problem has become far harder than you thought. 

It could be that you just made some silly modelling mistake (such as multiplying continuous variables and binary variables) or you simply have a problem now which is effectively unsolvable

**Reply by johanlofberg** [reply]
in other words, without details impossible to say anything else than the fact that you destroyed tractability of your model by the way you did it

**Reply by LiuJincsu** [reply]
Thank you for your answer. I rechecked my model and was able to solve it and get the result. But a prompt appears in the command bar:Warning: Solver not applicable (cplex).
May I ask if such a warning has any effect on my solution result? Will it seriously affect my solving accuracy?

**Reply by johanlofberg** [reply]
That statement makes no sense. You say the solver was not applicable (hence YALMIP did not solve the problem) but still you say you got a result.
